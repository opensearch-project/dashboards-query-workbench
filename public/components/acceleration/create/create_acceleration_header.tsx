/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiPageHeader,
  EuiPageHeaderSection,
  EuiTitle,
  EuiSpacer,
  EuiText,
  EuiLink,
} from '@elastic/eui';
import React from 'react';
import { OPENSEARCH_ACC_DOCUMENTATION_URL } from '../../../../common/constants';

export const CreateAccelerationHeader = () => {
  return (
    <div>
      <EuiPageHeader>
        <EuiPageHeaderSection>
          <EuiTitle size="l" data-test-subj="acceleration-header">
            <h1>Create Acceleration Index</h1>
          </EuiTitle>
        </EuiPageHeaderSection>
      </EuiPageHeader>
      <EuiSpacer size="s" />
      <EuiText size="s" color="subdued">
        Create OpenSearch Indexes from external data connections for better performance.{' '}
        <EuiLink external={true} href={OPENSEARCH_ACC_DOCUMENTATION_URL} target="_blank">
          Learn more
        </EuiLink>
      </EuiText>
    </div>
  );
};
