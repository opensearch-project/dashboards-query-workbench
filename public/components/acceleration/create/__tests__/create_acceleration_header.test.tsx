/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/dom';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { CreateAccelerationHeader } from '../create_acceleration_header';

describe('Acceleration header', () => {
  configure({ adapter: new Adapter() });

  it('renders acceleration flyout header', async () => {
    const wrapper = mount((<CreateAccelerationHeader />) as React.ReactElement<any>);
    wrapper.update();
    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
