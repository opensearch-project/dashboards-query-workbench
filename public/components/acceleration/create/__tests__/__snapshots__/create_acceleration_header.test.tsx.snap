// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Acceleration header renders acceleration flyout header 1`] = `
<div>
  <header
    className="euiPageHeader euiPageHeader--responsive euiPageHeader--center"
  >
    <div
      className="euiPageHeaderSection"
    >
      <h1
        className="euiTitle euiTitle--large"
        data-test-subj="acceleration-header"
      >
        Accelerate data
      </h1>
    </div>
  </header>
  <div
    className="euiSpacer euiSpacer--s"
  />
  <div
    className="euiText euiText--small"
  >
    <div
      className="euiTextColor euiTextColor--subdued"
    >
      Create OpenSearch Indexes from external data connections for better performance.
       
      <a
        className="euiLink euiLink--primary"
        href="https://opensearch.org/docs/latest"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn more
        <svg
          aria-hidden={true}
          aria-label="External link"
          className="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
          focusable="false"
          height={16}
          role="img"
          style={null}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        />
        <span
          className="euiScreenReaderOnly"
        >
          (opens in a new tab or window)
        </span>
      </a>
    </div>
  </div>
</div>
`;
